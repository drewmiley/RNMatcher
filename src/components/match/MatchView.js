import React, { Component } from 'react';
import {
    Text,
    View
} from 'react-native';

import VersusLabel from '../label/VersusLabel';

export default class MatchView extends Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = (nextProps, nextState) => {
            return this.props.matchPlayers !== nextProps.matchPlayers;
        };
    };
    render() {
        return <View>
            <Text>
                {this.props.matchPlayers.get(0).get('name')}
            </Text>
            <VersusLabel />
            <Text>
                {this.props.matchPlayers.get(1).get('name')}
            </Text>
        </View>
    };
};