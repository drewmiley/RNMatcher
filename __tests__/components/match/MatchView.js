import 'react-native';
import { List, Map } from 'immutable';
import React from 'react';
import renderer from 'react-test-renderer';

import MatchView from '../../../src/components/match/MatchView';

describe('MatchView', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
        	const matchPlayers = List.of(
                Map({name: 'One'}),
                Map({name: 'Two'})
            );
            const component = renderer.create(
                <MatchView matchPlayers={matchPlayers} />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});