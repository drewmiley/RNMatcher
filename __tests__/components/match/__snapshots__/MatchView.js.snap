exports[`MatchView Rendering should match the snapshot 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    One
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
     Versus 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Two
  </Text>
</View>
`;
